decidim-core/lib/decidim/commands/create_resource.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Commands
+    class CreateResource < ::Decidim::Command
+      include Decidim::Commands::ResourceHandler
+
+      # Initializes the command.
+      # @param form [Decidim::Form] the form object to create the resource.
+      def initialize(form)
+        @form = form
+      end
+
+      # Creates the result if valid.
+      #
+      # Broadcasts :ok if successful, :invalid otherwise.
+      def call
+        return broadcast(:invalid) if invalid?
+
+        transaction do
+          run_before_hooks
+          create_resource
+          run_after_hooks
+        end
+
+        broadcast(:ok, resource)
+      rescue Decidim::Commands::HookError
+        broadcast(:invalid)
+      end
+
+      protected
+
+      # @param soft [Boolean] whether to soft-create the resource or not.
+      # @usage
+      #  create_resource(soft: true) - Will soft-create the resource, returning any validation errors.
```

Comment:
This is used in other PRs, right?

Reply:
This is actually being used by assemblies...